### ISSUE TITLE
[stepfunctions]: Adjusting parameters in ContainerOverride of EcsRunTask

### ISSUE THREAD

**Issue by MieladN** [aws/aws-cdk#24057]
### Describe the bug

The EcsRunTask construct only accepts an array of strings as override commands. It cannot produce substitutable output like "Command.$": "$.commands" that's needed to read the override command from the execution input. This is a limitation of the CDK implementation, not of the ECS optimized integration itself. Using the 'CallAwsService' construct allows you to get around this, but the CallAwsService doesn't allow you to set the IntegrationPattern as RUN_JOB (.sync).

In simpler terms:

Using Amazon States language, you can set the ContainerOverride field for an EcsRunTask, for example, as

`"Command.$": "States.Array($.Foo,$.Bar,$.FooBar)"`

However, if you use CDK, this is not possible, because there is no method to adjust parameters in the ContainerOverride method.

CDK code will always generate:

`"Command": "States.Array($.Foo,$.Bar,$.FooBar)"`

missing the .$ after Command.

### Expected Behavior

What I expect is for there to be a parameter to adjust previously set parameters. 

Currently, this is not possible. The field only accepts JsonPath.listAt. However, it doesn't accept JsonPath.stringAt() or JsonPath.Array().

### Current Behavior

If you try to set the container field as 

```
   const FooBar = new tasks.EcsRunTask(this, 'Foobar', {
      containerOverrides: [{
         command: JsonPath.array(('States.Array($.Foo,$.Bar,$.Foo,$.Bar,$.Foo)'),
       }],
```

There'll be a red squiggly line under command, saying: 
Type 'string' is not assignable to type 'string[]'

If you put square brackets around it:

```
   const FooBar = new tasks.EcsRunTask(this, 'Foobar', {
      containerOverrides: [{
         command: [JsonPath.array(('States.Array($.Foo,$.Bar,$.Foo,$.Bar,$.Foo)')],
       }],
```

You get an error saying: 

''Cannot use JsonPath fields in an array, they must be used in objects''

### Reproduction Steps

.

### Additional Information/Context

If it wasn't clear, I asked about it here on StackOverflow:

[https://stackoverflow.com/questions/75313400/how-do-i-modify-parameters-in-a-container-overrides-section-of-a-step-functions](https://stackoverflow.com/questions/75313400/how-do-i-modify-parameters-in-a-container-overrides-section-of-a-step-functions)

### CDK CLI Version

2.63.2 

### Node.js Version

v19.4.0

### OS

MacOS

### Language

Typescript

**Comment by khushail** [comment]
Hi @MieladN , thanks for mentioning this. We are classifying this as a feature request as we don't support this right now. 
I am marking this issue as p2, which means that we are unable to work on this immediately.

We use +1s to help prioritize our work, and are happy to revaluate this issue based on community feedback. You can reach out to the [cdk.dev](http://cdk.dev/) community on [Slack](https://cdk.dev/) to solicit support for reprioritization.

**Comment by mKeRix** [comment]
I ran into the same issue just now and could create a workaround for myself by using the same method call that `JsonPath.listAt` makes underneath directly (circumventing its validator). In Golang it ends up looking like this: `awscdk.Token_AsList(awsstepfunctions.JsonPath_Array(jsii.String("$.foobar")))`. I believe in Typescript it should look like this in the end: `Token.asList(JsonPath.array("$.foobar"))`

**Comment by Belair34** [comment]
Also, for the record: OP had a concern about using CallAwsService as a workaround, but it not supporting .sync. The alternative to this is using a CustomState and for the "Resource" parameter specifying it like this: 
`Resource: 'arn:aws:states:::ecs:runTask.sync'`
This is another workaround, but I would go with the response above.

**Comment by deeTEEcee** [comment]
@mKeRix thanks for that, saved me some time from having to fall back to using CustomState again.
